migrations/src/main/scala/za/co/absa/enceladus/migrations/migrations/model1/Serializer1.scala

```diff
@@ -0,0 +1,123 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.migrations.migrations.model1
+
+import com.fasterxml.jackson.databind.{ObjectMapper, SerializationFeature}
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import io.github.cbartosiak.bson.codecs.jsr310.zoneddatetime.ZonedDateTimeAsDocumentCodec
+import org.bson.BsonDocumentWriter
+import org.bson.codecs.configuration.CodecRegistries
+import org.bson.codecs.configuration.CodecRegistries.{fromProviders, fromRegistries}
+import org.bson.codecs.{Codec, EncoderContext}
+import org.json4s.jackson.Serialization
+import org.json4s.{Formats, NoTypeHints}
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.bson.codecs.DEFAULT_CODEC_REGISTRY
+import org.mongodb.scala.bson.codecs.Macros._
+import za.co.absa.enceladus.migrations.migrations.model1.conformanceRule._
+
+/**
+  * This is the object for deserializing Model 1 version of Enceladus Schema
+  */
+object Serializer1 {
+  private val objectMapper = new ObjectMapper()
+    .registerModule(DefaultScalaModule)
+    .registerModule(new JavaTimeModule())
+    .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
+
+  implicit private val formatsJson: Formats = Serialization.formats(NoTypeHints).withBigDecimal
+
+  private val codecRegistry = fromRegistries(fromProviders(
+    classOf[Schema], classOf[SchemaField],
+    classOf[MappingTable], classOf[DefaultValue], classOf[MenasReference],
+    classOf[Dataset], classOf[ConcatenationConformanceRule],
+    classOf[MappingConformanceRule], classOf[LiteralConformanceRule], classOf[ConcatenationConformanceRule],
+    classOf[DropConformanceRule], classOf[SparkSessionConfConformanceRule], classOf[UppercaseConformanceRule],
+    classOf[SingleColumnConformanceRule], classOf[CastingConformanceRule], classOf[NegationConformanceRule]),
+    CodecRegistries.fromCodecs(new ZonedDateTimeAsDocumentCodec()), DEFAULT_CODEC_REGISTRY)
+
+  private val schemaCodec: Codec[Schema] = codecRegistry.get(classOf[Schema])
+  private val mappingTableCodec: Codec[MappingTable] = codecRegistry.get(classOf[MappingTable])
+  private val datasetCodec: Codec[Dataset] = codecRegistry.get(classOf[Dataset])
+
+  /**
+    * Serializes a Model 1 schema JSON
+    */
+  def serializeSchema(schema: Schema): String = {
+    val bsonDocument = new BsonDocument
+    val bsonWriter = new BsonDocumentWriter(bsonDocument)
+    val encodeContext = EncoderContext.builder.build
+
+    schemaCodec.encode(bsonWriter, schema, encodeContext)
+
+    bsonDocument.toJson
+
+    // alternative 1
+    //objectMapper.writeValueAsString(schema)
+    // alternative 2
+    //Serialization.write(schema)
```

Comment:
Are we keeping these?

Reply:
Will remove this. It only shows my struggle trying out different serializers to output `ZonedDataTime` in the extended format.